pkg/controller/node/controller_utils.go

```diff
@@ -37,16 +37,22 @@ import (
 
 // This function is expected to get a slice of NodeReadyConditions for all Nodes in a given zone.
 func ComputeZoneState(nodeReadyConditions []*api.NodeCondition) zoneState {
-	seenReady := false
+	readyNodes := 0
+	notReadyNodes := 0
 	for i := range nodeReadyConditions {
 		if nodeReadyConditions[i] != nil && nodeReadyConditions[i].Status == api.ConditionTrue {
-			seenReady = true
+			readyNodes++
+		} else {
+			notReadyNodes++
 		}
 	}
-	if seenReady {
-		return stateNormal
+	if readyNodes == 0 && notReadyNodes > 0 {
+		return stateFullDisruption
+	}
+	if notReadyNodes > 2 && 2*notReadyNodes > readyNodes {
+		return statePartialDisruption
```

Comment:
Please add a function comment that describes the algorithm that we are using to compute it, so that not everyone needs to understand the code.